### PR TITLE
Add expiration date validation for OpenRDT

### LINKED ISSUE #111: Confirm Expiration Date from QR Code
When we scan the QR code, we should be able to verify that it is okay to use via the expiration date.  If it is expired, it should create a toast message that it is expired, and return the user back to the patient screen to do another RDT.  I believe this already works for the CareStart test.  This needs to be implemented for Version 1.

### PR DESCRIPTION
**Pull request by vincent-karuri**
Closes #111

### CODE REVIEW

File: rdt-app/src/main/java/io/ona/rdt_app/widget/RDTBarcodeFactory.java [hunk 1]
```diff
@@ -95,20 +102,22 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
...
                                             }
                                         }
 
                                         // populate exp. date to expiration date widget value
                                         stepAndId = expirationDateAddress.isEmpty() ? new String[0] : expirationDateAddress.split(":");
                                         if (stepAndId.length == 2) {
-                                            jsonApi.writeValue(stepAndId[0].trim(), stepAndId[1].trim(), barcodeValues[1].trim(), "", "", "", false);
+                                            try {
+                                                expDate = convertDate(barcodeValues[1].trim(), OPEN_RDT_DATE_FORMAT);
+                                                jsonApi.writeValue(stepAndId[0].trim(), stepAndId[1].trim(), getDateStr(expDate), "", "", "", false);
```
Comment (githengi) [inline rdt-app/src/main/java/io/ona/rdt_app/widget/RDTBarcodeFactory.java]: Is the default Date format good enough. Why not use the ISO Date format for storing the date as a string

Comment (vincent-karuri) [reply rdt-app/src/main/java/io/ona/rdt_app/widget/RDTBarcodeFactory.java]: The date is picked up as is from the rdt qr code.

It is currently in this format.

Comment (githengi) [reply rdt-app/src/main/java/io/ona/rdt_app/widget/RDTBarcodeFactory.java]: Got that we get get the date from rdt using a custom format. But when saving on json we convert back to string using `date.toString();` using the method `getDateStr()`.

Comment (vincent-karuri) [reply rdt-app/src/main/java/io/ona/rdt_app/widget/RDTBarcodeFactory.java]: Ok, done.

**Pull request merged**